openquake/hazardlib/correlation.py

```diff
@@ -108,42 +136,199 @@ def __init__(self, vs30_clustering):
+        Returns the correlation model for a set of distances, given the
+        appropriate period
+
+        :param numpy.ndarray distances:
+            Distance matrix
+
+        :param float period:
+            Period of spectral acceleration
+        """
+        if isinstance(imt, SA):
+            period = imt.period
         else:
-            distances = sites_or_distances
+            assert isinstance(imt, PGA), imt
+            period = 0
 
         # formulae are from page 1700
-        if imt.period < 1:
-            if not vs30_clustering:
+        if period < 1:
+            if not self.vs30_clustering:
                 # case 1, eq. (17)
-                b = 8.5 + 17.2 * imt.period
+                b = 8.5 + 17.2 * period
             else:
                 # case 2, eq. (18)
-                b = 40.7 - 15.0 * imt.period
+                b = 40.7 - 15.0 * period
         else:
             # both cases, eq. (19)
-            b = 22.0 + 3.7 * imt.period
+            b = 22.0 + 3.7 * period
 
         # eq. (20)
         return numpy.exp((- 3.0 / b) * distances)
+
+    def get_lower_triangle_correlation_matrix(self, sites, imt, stddev_intra):
+        """
+        See :meth:`BaseCorrelationModel.get_lower_triangle_correlation_matrix`.
+        """
+        # Reshape 'stddev_intra' if needed.
+        stddev_intra_shape = stddev_intra.shape
+        try:
+            if stddev_intra_shape[0] > stddev_intra_shape[1]:
+                stddev_intra = numpy.transpose(stddev_intra)
+                stddev_intra = stddev_intra[0]
+        except:
+            stddev_intra = stddev_intra
+
+        COV = numpy.diag(stddev_intra) * \
+            self._get_correlation_matrix(sites, imt) * numpy.diag(stddev_intra)
+        return numpy.linalg.cholesky(COV)
+
+
+
+
+class HM2018CorrelationModel(BaseCorrelationModel):
+    """
+    "Uncertainty in intraevent spatial correlation of elastic pseudo-
+    acceleration spectral ordinates"
+    by Pablo Heresi and Eduardo Miranda. Submitted for possible publication
+    in Bulletin of Earthquake Engineering, 2018.
+
+    :param uncertainty_multiplier:
+        Value to be multiplied by the uncertainty in the correlation parameter
+        beta. If uncertainty_multiplier = 0, the median value is used as a
+        constant value.
+    """
+
+    def __init__(self, uncertainty_multiplier=0):
+        self.uncertainty_multiplier = uncertainty_multiplier
+        self.distance_matrix = {}
+        self.cache = {}
+
+    def set_distance_matrix(self, sites):
+        """
+        Calculate distance matrix for a given sites collection. The distance 
+        matrix is saved, so we do not calculate it every time.
+
+        :param sites:
+            :class:`~openquake.hazardlib.site.SiteCollection` residuals were
+            sampled for.
+        """
+        self.distance_matrix = sites.mesh.get_distance_matrix()
+
+    def get_lower_triangle_correlation_matrix(self, sites, imt):
+        pass
+
```

Comment:
Why this do nothing method?

Reply:
This is because the parent class defines this method, but it's not used in this new correlation class. I'll add a comment about it.